nixos/modules/services/web-apps/writefreely.nix

```diff
@@ -0,0 +1,102 @@
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.writefreely;
+  configFile = pkgs.writeText "config.ini" (generators.toINI {} cfg.settings);
+  settingsFormat = pkgs.formats.json {};
```

Comment:
these don't quite go together. is the config file json or ini? mixing generators and types like this can cause very hard to debug problems.